nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -458,156 +476,148 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final CsvPreference csvPref = getPreference(context, flowFile);
         final boolean header = context.getProperty(HEADER).asBoolean();
         final ComponentLog logger = getLogger();
-        final String schema = context.getProperty(SCHEMA).evaluateAttributeExpressions(flowFile).getValue();
-        final CellProcessor[] cellProcs = this.parseSchema(schema);
-        final boolean isWholeFFValidation = context.getProperty(VALIDATION_STRATEGY).getValue().equals(VALIDATE_WHOLE_FLOWFILE.getValue());
+        String schema = context.getProperty(SCHEMA).evaluateAttributeExpressions(flowFile).getValue();
```

Comment:
```suggestion
        final String schema = context.getProperty(SCHEMA).evaluateAttributeExpressions(flowFile).getValue();
```

Reply:
The schema cannot be final because if it's null, I modify it in line 519.

Reply:
Good point though instead of modifying it couldn't you just have
`cellProcs = this.parseSchema(newSchema.substring(0, newSchema.length() - 1));`

Reply:
I could, I just didn't want to make it confusing if down the line the schema variable needs to be used.